lib/auth/join_azure.go

```diff
@@ -145,7 +170,14 @@ func (cfg *azureRegisterConfig) CheckAndSetDefaults(ctx context.Context) error {
 	}
 	if cfg.getVMClient == nil {
 		cfg.getVMClient = func(subscriptionID string, token *azure.StaticCredential) (azure.VirtualMachinesClient, error) {
-			client, err := azure.NewVirtualMachinesClient(subscriptionID, token, nil)
+			opts := &armpolicy.ClientOptions{
+				ClientOptions: policy.ClientOptions{
+					Telemetry: policy.TelemetryOptions{
+						ApplicationID: azureUserAgent,
+					},
+				},
+			}
```

Comment:
What does this change do? Make it easier to detect when teleport is rate-limited? Is it a best-practice? something Azure told us to put there?

Can you add a note in the PR about additional changes?

Reply:
Yup, this should make it easier to identify teleport traffic. I'll add an additional note.